core/services/pipeline/common.go

```diff
@@ -0,0 +1,226 @@
+package pipeline
+
+import (
+	"database/sql/driver"
+	"encoding/json"
+	"net/url"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/pkg/errors"
+	"github.com/shopspring/decimal"
+
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/store/models"
+	"github.com/smartcontractkit/chainlink/core/utils"
+)
+
+//go:generate mockery --name Task --output ./mocks/ --case=underscore
+//go:generate mockery --name Config --output ./mocks/ --case=underscore
+
+type (
+	Task interface {
+		Type() TaskType
+		DotID() string
```

Comment:
What is this?

Reply:
In DOT, you give names to each node in the graph.  The DOT package I used calls this the "DOT ID".  I originally just discarded this info, but then I found several ways that it can be quite useful (not the least of which being that it makes some of the most important tests a hell of a lot easier to write)